modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/DDMTemplateBrowserSnifferUpgradeProcessTest.java

```diff
@@ -0,0 +1,164 @@
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+@RunWith(Arquillian.class)
+public class DDMTemplateBrowserSnifferUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_addDDMTemplateContent();
+	}
+
+	@Test
+	public void testUpgradeDDMTemplateRemoveBrowserSniffer() throws Exception {
+		List<DDMTemplate> ddmTemplateList = _addDDMTemplateContent();
+
+		for (DDMTemplate ddmTemplateIndex : ddmTemplateList) {
+			long templateId = ddmTemplateList.get(
+				(int)ddmTemplateIndex.getTemplateId()
+			).getTemplateId();
+
+			_runUpgrade();
+
+			String scriptResult = _ddmTemplateService.getTemplate(
+				templateId
+			).getScript();
+
+			Assert.assertEquals(
+				_read("updated-ddm-template-browser-sniffer-content.ftl"),
+				scriptResult);
+		}
+	}
+
+	private List<DDMTemplate> _addDDMTemplateContent() throws Exception {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		long classPK = ParamUtil.getLong(serviceContext, "classPK");
+
+		_ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL,
+			_read("ddm-template-browser-sniffer-content.ftl"),
+			serviceContext);
+
+		return _ddmTemplateService.getTemplates(
+			_group.getCompanyId(), _group.getClassNameId(),
+			_portal.getClassNameId(DDMTemplate.class),
+			_portal.getClassNameId(JournalArticle.class), 1);
+	}
+
+	private String _read(String fileName) throws Exception {
+		Class<?> clazz = getClass();
+
+		return StringUtil.read(
+			clazz.getClassLoader(),
+			"com/liferay/dynamic/data/mapping/dependencies/upgrade.v5_3_2/" +
+				fileName);
+	}
+
+	private void _runUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME, LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
```

Comment:
@alberttgomes, is this really needed? Can you please clarify?

Reply:
I based on this implementation [link](https://github.com/liferay/liferay-portal/blob/7.4.3.73-ga73/modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/internal/upgrade/v4_4_3/test/JournalArticleLayoutClassedModelUsageUpgradeProcessTest.java#L399) listed on this [documentation](https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Testing) But, I believe which be to for clear cache. Can remove!